### DISCUSSION TITLE
Too many error messages produced for Discriminated Union Type validation

### DISCUSSION THREAD

**Discussion by charlesxucheng** [pydantic/pydantic#3677] [Question] [answered]
Hi. I was trying out the tagged union feature added to 1.9.0. Based on the example in the doc: https://pydantic-docs.helpmanual.io/usage/types/#discriminated-unions-aka-tagged-unions, I created the following model:

```python
class Cat(BaseModel):
    type: Literal['cat']
    meows: int


class Dog(BaseModel):
    type: Literal['dog']
    barks: float


class Lizard(BaseModel):
    type: Literal['reptile', 'lizard']
    scales: bool


class Model(BaseModel):
    pet: Union[Cat, Dog, Lizard] = Field(..., discriminator='type')
    id: int


class PetsModel(BaseModel):
    name: str
    pets: List[Model]
```

And the test code:
```python
import yaml
from manifest import PetsModel


class TestManifest2:
    def test_instantiate(self):
        manifest = yaml.full_load('''
        name: TEST
        pets:
          - pet: 
              type: cat
              meows: 1
            id: 1
          - pet:
              type: dog
              meows: 1
            id: 2
        ''')
        result = PetsModel(**manifest)
        print(result)
        assert isinstance(result, PetsModel)
```

The test failed as per expected. However, why there are 4 errors? I expect only one error saying the attribute barks is missing for the second pet with type dog but have 'meows'. One 'expected value' error will be produced for each union type, so the list of error will be very long in cases where there are quite a lot of possible types in the union type (in my case there will be dozens), and that will make it hard to pinpoint the exact error. I wonder if there is a way to reduce the number of errors produced?

E   pydantic.error_wrappers.ValidationError: 4 validation errors for PetsModel
E   pets -> 1 -> pet -> type
E     unexpected value; permitted: 'cat' (type=value_error.const; given=dog; permitted=('cat',))
E   pets -> 1 -> pet -> barks
E     field required (type=value_error.missing)
E   pets -> 1 -> pet -> type
E     unexpected value; permitted: 'reptile', 'lizard' (type=value_error.const; given=dog; permitted=('reptile', 'lizard'))
E   pets -> 1 -> pet -> scales
E     field required (type=value_error.missing)

**Comment by PrettyWood** [comment, marked as answer]
Hi @charlesxucheng 
I can't reproduce this with pydantic 1.9. I get the following error
```
pydantic.error_wrappers.ValidationError: 1 validation error for PetsModel
pets -> 1 -> pet -> Dog -> barks
```

**Reply by charlesxucheng** [reply]
Thanks @PrettyWood . My bad. I found my Pydantic version still at 1.8.2. I was changing the version to do some troubleshooting and messed it up. After I changed to 1.9.0, the error is as per expected. Sorry for the false alarm.
